src/components/elements/StepDialog.tsx

```diff
@@ -0,0 +1,189 @@
+import { LoadingButton } from '@mui/lab';
+import {
+  Box,
+  Button,
+  ButtonProps,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Stack,
+  Typography,
+} from '@mui/material';
+import { ArrowLeftIcon, ArrowRightIcon } from '@mui/x-date-pickers';
+import React, { ReactNode, useMemo, useState } from 'react';
+import ButtonTooltipContainer from '@/components/elements/ButtonTooltipContainer';
+import CommonDialog, {
+  CommonDialogProps,
+} from '@/components/elements/CommonDialog';
+import Loading from '@/components/elements/Loading';
+
+export type StepDefinition = {
+  title: string; // must be unique, since it is used as the step's key
+  content: ReactNode;
+
+  // if onSubmit is not provided, the default action is to just go to the next step
+  onSubmit?: () => Promise<any>; // after onSubmit promise resolves, go to the next step if there is one
+  submitButtonText?: string;
+  submitLoading?: boolean;
+  ButtonProps?: ButtonProps;
+  omitStepTitle?: boolean;
```

Comment:
For more flexibility, we could replace:
```ts
title: string; // must be unique, since it is used as the step's key
omitStepTitle?: boolean;
```
with
```ts
key: string // unique key for the step
title?: string // in the future we could expand this to ReactNode for more flexibility. it is a presentation prop
```

That would let us expand `title` to accept a ReactNode in the future if we need more flexibility in rendering. It keeps `title` presentation-focused while `key` is logical, and we non longer need a prop for hiding the title.